src/aiovodafone/api.py

```diff
@@ -38,6 +38,28 @@ class VodafoneStationDevice:
 class VodafoneStationCommonApi(ABC):
     """Common API calls for Vodafone Station routers."""
 
+    @staticmethod
+    async def get_device_type(host: str) -> DeviceType:
+        """Finds out the device type of a Vodafone Stations and returns it as enum.
+        The Technicolor devices always answer with a valid HTTP response, the
+        Sercomm returns 404 on a missing page. This helps to determine which we are
+        talking with.
+
+        Args:
+            host (str): The router's address, e.g. `192.168.1.1`
+
+        Returns:
+            DeviceType: If the device is a Technicolor, it returns
+            `DeviceType.TECHNICOLOR` otherwise `DeviceType.SERCOMM`.
+        """
+        async with aiohttp.ClientSession() as session:
```

Comment:
This should accept an aiohttp session

https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=plat#platinum-

Reply:
I agree, I can change that.  We should then also accept a Client session for the API in general. Right now, it creates the session in the constructor:
https://github.com/chemelli74/aiovodafone/blob/386ff478927a8d2912236e49e4d855f7e25ffb2d/src/aiovodafone/api.py#L55

Reply:
I added a parameter incl. documentation to accept a `ClientSession`.